### PR TITLE
Add mDNS capability

### PR DESCRIPTION
**Pull request by krzmaz**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -122,6 +123,10 @@ int main() {
         printf("IP Address: %lu.%lu.%lu.%lu\n", ip_addr & 0xFF, (ip_addr >> 8) & 0xFF, (ip_addr >> 16) & 0xFF, ip_addr >> 24);
     }
 
+    const auto hostname =  "pico-w-neopixel-server";
+    mdns_resp_init();
+    mdns_resp_add_netif(&cyw43_state.netif[CYW43_ITF_STA], hostname);
```
Comment (mrozycki) [inline src/main.cpp:128]: Since it's hardcoded, what happens if you have 2 running on the same network?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,10 @@
 set(PROGRAM_NAME pico_w_neopixel_server)
 
+# TODO fix after https://github.com/georgerobotics/cyw43-driver/pull/39 gets into the SDK
+set(BROKEN_FILE ${PICO_CYW43_DRIVER_PATH}/src/cyw43_lwip.c)
+file(READ ${BROKEN_FILE} FILE_CONTENTS)
+string(REPLACE "LWIP_MDNS_RESPONDER" "LWIP_1_MDNS_RESPONDER" FILE_CONTENTS "${FILE_CONTENTS}")
+file(WRITE ${BROKEN_FILE} "${FILE_CONTENTS}")
```
Comment (mrozycki) [inline src/CMakeLists.txt]: 1. `BROKEN_FILE` is not a great name.
2. Re TODO: fix how?

Comment (krzmaz) [reply src/CMakeLists.txt]: 1. Changed to `FILE_TO_CHANGE` and added some more explanation what is done there
2. Replaced `fix` with `remove`

File: src/main.cpp (hunk 1 above)
Comment (krzmaz) [reply src/main.cpp:128]: The [RFC](https://www.rfc-editor.org/rfc/rfc6762.html#page-31) suggests changing the hostname by appending an index, but the [lwIP code](https://github.com/lwip-tcpip/lwip/blob/master/src/apps/mdns/mdns.c#L1789) doesn't implement it, and neither did I.

I've added a warning about this in the README. In the future, hostnames should be configurable via the config endpoint, see https://github.com/krzmaz/pico-w-webserver-example/milestone/1

**Review by mrozycki — APPROVED** [review]

**Comment by mrozycki** [comment]
Confirmed works on windows 10 (and wsl)

**Pull request merged**